byterun/debugger.c

```diff
@@ -194,11 +195,15 @@ void caml_debugger_init(void)
     /* Unix domain */
     sock_domain = PF_UNIX;
     sock_addr.s_unix.sun_family = AF_UNIX;
+    a_len = strlen(address);
+    if (a_len >= sizeof(sock_addr.s_unix.sun_path)) {
+      caml_fatal_error("Debug socket path length exceeds maximum permitted length");
+    }
     strncpy(sock_addr.s_unix.sun_path, address,
-            sizeof(sock_addr.s_unix.sun_path));
+            sizeof(sock_addr.s_unix.sun_path) - 1);
```

Comment:
There is still an off-by-one error here: if `strlen(address) == sizeof(sock_addr.s_unix.sun_path) - 1`, then the fatal error won't trigger and `strncpy` will fail to write a null terminator. I think the check should be `a_len >= sizeof(sock_addr.s_unix.sun_path) - 1`.

Reply:
But `strlen(address) == sizeof(sock_addr.s_unix.sun_path) - 1` should be a valid length since it fits into the buffer. Perhaps we should just explicitly set the final byte to `0` -- I'd assumed that the buffer was zeroed already and so the `strncpy` didn't need to set the final byte to 0.

Reply:
Yeah, the issue is that `strncpy` won't write the final zero. Setting it to `0` explicitly works.